File: OOP/3-Extension-Methods-Delegates-Lambda-LINQ/03-05.SelectStudents/Student.cs
using System;

namespace SelectStudents
{
    /// <summary>
    /// Represents a student.
    /// </summary>
    public struct Student
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public int Age { get; set; }

        /// <summary>
        /// Initializes new instance of student struct to a specified first name, last name and age.
        /// </summary>
        /// <param name="firstName">First name of the student.</param>
        /// <param name="lastName">Last name of the student.</param>
        /// <param name="age">Age of the student.</param>
        public Student(string firstName, string lastName, int age) : this()
        {
            this.FirstName = firstName;
            this.LastName = lastName;
            this.Age = age;
        }
    }
}

File: JavaScript-Applications/1-Dive-Into-Underscore-JS/scripts/data/student.js
define([], function() {
	'use strinct'

	var Student = (function () {
		function Student (firstName, lastName, age, marks) {
			this.setFirstName(firstName);
			this.setLastName(lastName);
			this.setAge(age);
			this.setMarks(marks);
		}

		Student.prototype = {
			getFullName: function () {
				return this._firstName + ' ' + this._lastName;
			},
			setFirstName: function (firstName) {
				if (typeof firstName !== 'string') {
					throw { message: 'Student\' first name must be string!' };
				}

				this._firstName = firstName;
			},
			getFirstName: function () {
				return this._firstName;
			},
			setLastName: function (lastName) {
				if (typeof lastName !== 'string') {
					throw { message: 'Student\'s last name must be string!' };
				}

				this._lastName = lastName;
			},
			getLastName: function () {
				return this._lastName;
			},
			setAge: function (age) {
				if (typeof age !== 'number') {
					throw { message: 'Student\'s age must be a number!' };
				}

				if (age < 1) {
					throw { message: 'Student\'s age must be positive number!' };
				}

				this._age = age;
			},
			getAge: function () {
				return this._age;
			},
			setMarks: function (marks) {
				if (!(marks instanceof Array)) {
					throw { message: 'Student\'s marks must be an array' };
				}

				if (marks.length < 1) {
					throw { message: 'Student\'s marks must be positive number'}
				}

				this._marks = marks;
			},
			getMarks: function () {
				return this._marks;
			},
			toString: function () {
				// return JSON.stringify(this);
				return 'FirstName: ' + this._firstName + ', ' +
						'LastName: ' + this._lastName + ', ' +
						'Age: ' + this._age + ', ' +
						'Marks: ' + this._marks;
			}
		};

		return Student;
	})();

	return Student;
});